gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/algorithm/generator/AbstractGeneratorTest.java

```diff
@@ -58,11 +58,23 @@ protected boolean same(final Graph g1, final Graph g2) {
 
     private boolean sameInVertices(final Vertex v, final List<Vertex> inVertices) {
         return inVertices.stream()
-                .allMatch(inVertex -> IteratorUtils.filter(v.vertices(Direction.IN), hv -> hv.value("oid").equals(inVertex.value("oid"))).hasNext());
+                .allMatch(inVertex ->
+                {
+                    final Iterator<Vertex> itty = v.vertices(Direction.IN);
+                    final boolean result = IteratorUtils.filter(itty, hv -> hv.value("oid").equals(inVertex.value("oid"))).hasNext();
+                    CloseableIterator.closeIterator(itty);
```

Comment:
Can we do this in a try/finally everywhere so we don't leak resources in exception cases?